### PR TITLE
chore: bump celestia-app to v1.0.0-rc2

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

bumps app to latest release candidate. I'm not sure how celestia-node wants to handle app version changes. I'm completely unaware of any existing plans, so aplogies if this is already planned for, but ideally we don't rely on the latest appversion constant. This is definitely fine for v1, but when v2 comes along we will want to change which version we use according to what is in the header.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: share/share.go [hunk 1]
```diff
@@ -18,15 +18,18 @@ var (
...
 	// NamespaceSize is a system-wide size for NMT namespaces.
 	NamespaceSize = appconsts.NamespaceSize
 	// Size is a system-wide size of a share, including both data and namespace ID
 	Size = appconsts.ShareSize
 )
 
+var (
+	// MaxSquareSize is currently the maximum size supported for unerasured data in
+	// rsmt2d.ExtendedDataSquare.
+	MaxSquareSize = appconsts.SquareSizeUpperBound(appconsts.LatestVersion)
```
Comment (renaynay) [inline share/share.go:30]: I wonder if we can just set this to ipld.MaxSquareSize

**Review by renaynay — DISMISSED** [review]

**Review by vgonkivs — DISMISSED** [review]

**Review by renaynay — APPROVED** [review]

**Review by Wondertan — DISMISSED** [review]
Approving!

We need to cleanup the usage of this constant, thus https://github.com/celestiaorg/celestia-node/issues/2309

File: share/share.go (hunk 1 above)
Comment (evan-forbes) [reply share/share.go]: https://github.com/celestiaorg/celestia-node/pull/2305/commits/4719e5130a6e685048470cd140cba0d2765f27b2

**Comment by Wondertan** [comment]
@evan-forbes, linter seems to complain after the merge, do you mind fixing it?

**Comment by evan-forbes** [comment]
sorry forgot to mention @Wondertan fixed in https://github.com/celestiaorg/celestia-node/pull/2305/commits/46e2a70f918ed97c19d9600b65bcf66b17ba8b03

also bumped to rc2 since why not

**Review by Wondertan — APPROVED** [review]

**Review by distractedm1nd — APPROVED** [review]

**Review by walldiss — APPROVED** [review]

**Review by walldiss — APPROVED** [review]

**Review by walldiss — APPROVED** [review]

**Pull request merged**